frontend/src/Patients/PatientProfileModal.jsx

```diff
@@ -0,0 +1,128 @@
+        setSelectedAppointment(null);
+        alert('Appointment Updated Successfully');
+      } else {
+        const response = await api.post('/api/user/appointments', { ...appointmentData, patientId: patient.id }, { withCredentials: true });
+        setAppointments([...appointments, response.data]);
+        alert('Appointment Created Successfully');
+      }
+      setApptModalOpen(false);
+    } catch (error) {
+      console.error('Error creating appointment:', error.response ? error.response.data : error.message);
+      alert('An error occurred while creating the appointment. Please try again.');
+    }
+  };
+
+  const handleEditAppointment = (appointment) => {
+    setSelectedAppointment(appointment);
+    setApptModalOpen(true);
+  };
+
+  const handleDeleteAppointment = async (id) => {
+    try {
+      await api.delete(`/api/user/appointments/${id}`, { withCredentials: true });
+      setAppointments(appointments.filter(appointment => appointment.id !== id));
+    } catch (error) {
+      console.error('Error deleting appointment:', error.response ? error.response.data : error.message);
+    }
+  };
+
+  if (!isOpen) return null;
+
+  return (
+    <div className="modal-overlay">
+      <div className="modal-content">
+        <button className="close-button" onClick={onClose}>Close</button>
+        {patient && (
+          <>
+            <ProfileCard patient={patient} setPatient={setPatient} />
+            <div className='edit-buttons'>
+              <button onClick={() => setProfileModalOpen(true)}>Edit Profile</button>
+              <button onClick={handleDeletePatient}>Delete Profile</button>
+            </div>
+            <ProfileModal
+              isOpen={isProfileModalOpen}
+              onClose={() => setProfileModalOpen(false)}
+              handleSubmitPatientInfo={handleEditPatient}
+              initialData={patient}
+              title="Edit Patient Profile"
+            />
+            <div className='appointments'>
+              <div className='createApptButton'>
+                <button onClick={() => setApptModalOpen(true)}>Add Appointment</button>
+              </div>
+              <ApptModal
+                isApptModalOpen={isApptModalOpen}
+                handleApptModalClose={() => setApptModalOpen(false)}
+                handleSubmitAppointment={handleSubmitAppointment}
+                title={selectedAppointment ? "Edit Appointment" : "Create Appointment"}
+                initialData={selectedAppointment}
+              />
+              <div className='apptList'>
+                <div className='apptHeadline'>
+                  <h3>Appointments</h3>
+                </div>
+                {appointments.map((appointment) => (
+                  <ApptCard key={appointment.id} appointment={appointment} handleDeleteAppointment={handleDeleteAppointment} onEdit={handleEditAppointment} />
```

Comment:
Curious -- will these appointments be displayed in order? i.e. with soonest appointment at the top

Reply:
Also this might be a to-do for later, but will there be a place to view a list previous appointments? (or are they all displayed here?)

Reply:
Oh, yes, I used the Date & time of the appointments to arrange the soonest on top. Also, I made it such that the past appointments are greyed out and moved to the bottom.

Reply:
if it's similar logic, can we try to centralize the code into a general appointments file that can be utilized by both the patient and physician?

Reply:
I don't completely understand how that can be implemented. Can you elaborate a bit more?

Reply:
nvm, misunderstood your comment! i was thinking something similar to matt's comment above - moving the appointment helper functions into their own file potentially